src/core/thread.d

```diff
@@ -2293,7 +2294,7 @@ extern (C) void thread_joinAll()
             Thread.remove(t);
             t = tn;
         }
-        else if (t.isDaemon)
```

Comment:
Now the check whether sth. is a daemon thread is no longer synchronized w/ setting this flag. But in fact the old synchronization wasn't correct either. It's Thread.join that must synchronize w/ daemonizing a thread, in order to eliminate any chance for join to wait infinitely on a daemon thread.